app/script/controllers/product.list.controller.js

```diff
@@ -1,19 +1,29 @@
 (function () {
     angular.module('app')
         .controller('productListController', function ($http, $scope, $locale, apiService, $location) {
-              apiService.getProducts()
+            apiService.getProducts()
                     .then(function (response) {
-                         $scope.products=response.data;
+                        $scope.products = [];
+                        $scope.allProducts = [];
```

Comment:
Why is it an array of arrays? It should be just array of the products (response.data) and you won't need the ng-repeat